x-pack/plugins/security_solution/public/cases/components/case_view/index.tsx

```diff
@@ -281,6 +331,33 @@ export const CaseComponent = React.memo<CaseProps>(
       [allCasesLink]
     );
 
+    const showAlert = useCallback(
+      (alertId: string, index: string) => {
+        dispatch(
+          timelineActions.toggleExpandedEvent({
+            timelineId: TimelineId.active,
+            event: {
+              eventId: alertId,
+              indexName: index,
+              loading: false,
+            },
+          })
+        );
+      },
+      [dispatch]
+    );
+
+    useEffect(() => {
+      dispatch(
+        timelineActions.toggleExpandedEvent({
+          timelineId: TimelineId.active,
+          event: {},
+        })
+      );
+      // We need to close the EventFlyout on component mount in case it was open before in detections
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I started to allow the dependency and just create a useRef for initialization
```
const init = useRef(true)

useEffect(() => {
  if(init.current) {
    init.current = false;
    // Do what you have to do
 }
```
what do you think?

Reply:
I like it!